.github/workflows/pr.yml

```diff
@@ -14,6 +14,8 @@ concurrency:
 
 jobs:
   pr:
+    env:
+      dockerhub_username: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment:
Trying the pattern documented here https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-secrets to see if it allows us to conditionally execute a step if the secret is available.